contracts/transfer/circuits/src/execute/tests.rs

```diff
@@ -0,0 +1,341 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::helpers::{NoteLeaf, FETCH_PP_FROM_RUSK_PROFILE};
+use crate::ExecuteCircuit;
+
+use anyhow::{anyhow, Result};
+use canonical_host::MemStore;
+use dusk_pki::{Ownable, SecretSpendKey};
+use phoenix_core::Note;
+use poseidon252::tree::{PoseidonAnnotation, PoseidonTree};
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+
+use dusk_plonk::prelude::*;
+
+use std::convert::TryInto;
+
+macro_rules! test_execute {
```

Comment:
Is a macro really needed here? Can't you just use a function to do the same job and then the proper test?

Reply:
A macro is the shortest solution because we have different  inputs to these test, some of them being const generics and others usize inputs.

With a macro, we reduce a few lines repeatable function to

```
test_execute!(execute_1_0, 15, 1, 0);
```

With this one, we define with a single line the test name, the const generic CAPACITY to `15`, the inputs to `1` and outputs to `0`

Reply:
Fixed